AdyenTests/Adyen Tests/Components/APIClientMock.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  AnyRetryAPIClientMock.swift
+//  AdyenTests
+//
+//  Created by Mohamed Eldoheiri on 8/12/20.
+//  Copyright © 2020 Adyen. All rights reserved.
+//
+
+import Foundation
+@testable import Adyen
+
+enum DummyError: Error {
+    case dummy
+}
+
+final class APIClientMock: APIClientProtocol {
+
+    var mockedResponse: Response?
+
+    var mockedError: Error?
```

Comment:
Maybe we can use a mocked sequence of network results, so we can also cover cases with multiple re-tries and "pending-pending-error" scenarios ?

Something like:

```
internal enum NetworkResult {
    case .responce(Responce)
    case .error(Error)
}

final class APIClientMock: APIClientProtocol {
    var mockedResults: [NetworkResult] = []
...
```

Reply:
Thats a good idea actually, Thanks!

Reply:
done